includes/service-providers/constant_contact/class-newspack-newsletters-constant-contact-sdk.php

```diff
@@ -622,4 +625,138 @@ public function upsert_contact( $email_address, $data = [] ) {
 		);
 		return $this->get_contact( $email_address );
 	}
+
+	/**
+	 * Fetches a tag by its name
+	 *
+	 * Constant Contact API does not offer an endpoint to search for a tag by its name, so we have to query for all tags and look through them.
```

Comment:
Doesn't sound like there's anything we can do about this right now, but does this mean it's possible to create multiple lists with the same or similar names and then have this method fetch the wrong one? e.g.

My Tag 1
My tag

Searching for the second tag might return the first one if it appears in the response first.

The risk seems low, but we might want to consider some help text on the WP side to warn users about this. Maybe in a future PR

Reply:
Alternatively, it appears that the [endpoint to create tags](https://developer.constantcontact.com/api_guide/tags_create.html#) in the CC API returns the ID in the response. Could we store that ID as post meta and query using that, with the querying by name used only as a fallback?

Reply:
> Doesn't sound like there's anything we can do about this right now, but does this mean it's possible to create multiple lists with the same or similar names and then have this method fetch the wrong one? e.g.
> 
> My Tag 1 My tag
> 
> Searching for the second tag might return the first one if it appears in the response first.
> 
> The risk seems low, but we might want to consider some help text on the WP side to warn users about this. Maybe in a future PR

No, the search is for the exact full name. The only risk is that if you have more than 1000 tags in your account, we might not find an existing tag. We might then try to create it (assuming it doesn't already exists) but then the API will return an error saying the tag already exists - and they don't give us the ID, only the error message... so we end up in a dead end.

We could suggest them to add the existing ID to the error message when you try to create a tag that already exists.

> Alternatively, it appears that the [endpoint to create tags](https://developer.constantcontact.com/api_guide/tags_create.html#) in the CC API returns the ID in the response. Could we store that ID as post meta and query using that, with the querying by name used only as a fallback?

That's what we do! The only time we do that Search is when you are configuring a list to the provider for the first time and we check if a tag with that name already exists to use it instead of creating a new one.